### PR TITLE
Add typedoc documentation

### PR DESCRIPTION
**Pull request by vblagoje**
Adding typedoc documentation. I had to make a change in src/services/domain/IDomain.ts for the typedoc to work. It caught the type assignment mismatch that I fixed. Have a look @ryanemerson

### CODE REVIEW

File: src/services/domain/IDomain.ts [hunk 1]
```diff
@@ -1,6 +1,5 @@
-import {IServerAddress} from "../server/IServerAddress";
 
 export interface IDomain {
   controller: string;
-  servers: IServerAddress[]; // Map of format hostname: [server1, server2]
+  servers: string[][]; // Map of format hostname: [server1, server2]
```
Comment (ryanemerson) [inline src/services/domain/IDomain.ts]: Why do we need this change?

Comment (vblagoje) [inline src/services/domain/IDomain.ts]: Typedoc flags it as an incompatible assignment. I tested in a debugger and it indeed is. We are not dealing here with IServerAddress types but with an array of string arrays. Please see it for your self and check the types

File: gulpfile.js [hunk 2]
```diff
@@ -40,6 +41,18 @@ gulp.task('serve:dist', ['build:dist'], require('./tasks/server')(gulp, config.d
...
+    .src(['src/services/**/*.ts'])
+    .pipe(typedoc({
+      module: 'commonjs',
+      target: 'es5',
+      out: 'docs/',
+      name: 'Infinispan management console'
+    }))
+    ;
+});
+
```
Comment (ryanemerson) [inline gulpfile.js]: This method should be in its own file like the other gulp tasks. `gulpfile.js` is just an entry point.

File: gulpfile.js [hunk 3]
```diff
@@ -40,6 +41,18 @@ gulp.task('serve:dist', ['build:dist'], require('./tasks/server')(gulp, config.d
 gulp.task('serve', ['compile:src', 'less', 'fonts', 'html', 'json', 'assets'],
     require('./tasks/server')(gulp, config.srcDir, config.watchDir, true, config.projectDir, config));
 
+gulp.task('typedoc', function() {
+  return gulp
+    .src(['src/services/**/*.ts'])
```
Comment (ryanemerson) [inline gulpfile.js]: Why don't we just create the docs for all of our src files? That way we have everything

Comment (vblagoje) [inline gulpfile.js]: It gets really messy/busy and I am not sure we need those. Service API is the most important part IMHO. I am open to including if there is a decent argument for it.

File: gulpfile.js (hunk 2 above)
Comment (vblagoje) [inline gulpfile.js]: Ok +1

File: gulpfile.js (hunk 3 above)
Comment (ryanemerson) [inline gulpfile.js]: No argument, I was just curious

**Comment by vblagoje** [comment]
Have a look now @ryanemerson

**Comment by ryanemerson** [comment]
Integrated https://github.com/infinispan/infinispan-management-console/commit/ec695b2ff72b824aa193348d5d644bb2574fceab, thanks @vblagoje

Btw, I noticed that we didn't clean the docs dir so I added [this](https://github.com/infinispan/infinispan-management-console/blob/ec695b2ff72b824aa193348d5d644bb2574fceab/tasks/clean.js#L29)

**Pull request closed without merging**